test_conformance/extensions/cl_khr_command_buffer/command_buffer_event_sync.cpp

```diff
@@ -0,0 +1,1068 @@
+    {
+        log_error("EventMode::RET_WAIT_FOR_SEC_QUEUE_EVENT test with in order "
+                  "queue failed\n");
+        return TEST_FAIL;
+    }
+
+    if (TEST_PASS
+        != MakeAndRunTest<CommandBufferEventSync<
+            EventMode::RET_WAIT_FOR_SEC_QUEUE_EVENT, true>>(
+            device, context, queue, num_elements))
+    {
+        log_error("EventMode::RET_WAIT_FOR_SEC_QUEUE_EVENT test with out of "
+                  "order queue failed\n");
+        return TEST_FAIL;
+    }
+    return TEST_PASS;
+}
+
+//--------------------------------------------------------------------------
+// user-events test cases
+
+int test_user_event_wait(cl_device_id device, cl_context context,
+                         cl_command_queue queue, int num_elements)
+{
+    if (TEST_PASS
+        != MakeAndRunTest<
+            CommandBufferEventSync<EventMode::USER_EVENT_WAIT, false>>(
+            device, context, queue, num_elements))
+    {
+        log_error(
+            "EventMode::USER_EVENT_WAIT test with in order queue failed\n");
+        return TEST_FAIL;
+    }
+
+    if (TEST_PASS
+        != MakeAndRunTest<
+            CommandBufferEventSync<EventMode::USER_EVENT_WAIT, true>>(
+            device, context, queue, num_elements))
+    {
+        log_error(
+            "EventMode::USER_EVENT_WAIT test with out of order queue failed\n");
+        return TEST_FAIL;
+    }
+    return TEST_PASS;
+}
+
+int test_user_events_wait(cl_device_id device, cl_context context,
+                          cl_command_queue queue, int num_elements)
+{
+    if (TEST_PASS
+        != MakeAndRunTest<
+            CommandBufferEventSync<EventMode::USER_EVENTS_WAIT, false>>(
+            device, context, queue, num_elements))
+    {
+        log_error(
+            "EventMode::USER_EVENTS_WAIT test with in order queue failed\n");
+        return TEST_FAIL;
+    }
+
+    if (TEST_PASS
+        != MakeAndRunTest<
+            CommandBufferEventSync<EventMode::USER_EVENTS_WAIT, true>>(
+            device, context, queue, num_elements))
+    {
+        log_error("EventMode::USER_EVENTS_WAIT test with out of order queue "
+                  "failed\n");
+        return TEST_FAIL;
+    }
+    return TEST_PASS;
+}
+
+int test_user_event_callback(cl_device_id device, cl_context context,
+                             cl_command_queue queue, int num_elements)
+{
+    if (TEST_PASS
+        != MakeAndRunTest<
+            CommandBufferEventSync<EventMode::USER_EVENT_CALLBACK, false>>(
+            device, context, queue, num_elements))
+    {
+        log_error(
+            "EventMode::USER_EVENT_CALLBACK test with in order queue failed\n");
+        return TEST_FAIL;
+    }
+
+    if (TEST_PASS
+        != MakeAndRunTest<
+            CommandBufferEventSync<EventMode::USER_EVENT_CALLBACK, true>>(
+            device, context, queue, num_elements))
+    {
+        log_error("EventMode::USER_EVENT_CALLBACK test with out of order queue "
```

Comment:
On an implementation which does support command-buffers, but only in-order queues, I'm seeing a fail here rather than a when `MakeAndRunTest` returns `TEST_SKIPPED_ITSELF`

Reply:
please let me know if this is still the case (after applying new pattern of the test)